### PR TITLE
Make the Changes to prevent Duplicates from wrong FacebookID

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#68031](https://central.tri.be/issues/68031)

### CODE REVIEW

File: src/Tribe/Aggregator/Event.php [hunk 1]
```diff
@@ -158,11 +158,22 @@ public function get_existing_ids( $origin, $values ) {
...
 
-		return $wpdb->get_results( $wpdb->prepare( $sql, $key ), OBJECT_K );
+		/**
+		 * Allows us to check for legacy meta keys
+		 */
+		if ( ! empty( $fields[ $origin ]['legacy'] ) ) {
+			$keys[] = $key;
+			$keys[] = "_{$fields[ $origin ]['legacy']}";
+
+			$sql .= 'AND meta_key IN ( "' . implode( '", "', $keys ) .'" )';
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Event.php]: Worth passing `$keys` through `esc_sql()` first, or safe as is?

File: src/Tribe/Aggregator/Event.php [hunk 2]
```diff
@@ -158,11 +158,22 @@ public function get_existing_ids( $origin, $values ) {
...
+		} else {
+			$sql .= 'AND meta_key = ' . esc_sql( $key );
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Event.php]: Should this also be single quoted besides being escaped?

File: src/Tribe/Aggregator/Event.php (hunk 1 above)
Comment (bordoni) [inline src/Tribe/Aggregator/Event.php]: We could. Would make it safer, just gotta change the `implode`/

File: src/Tribe/Aggregator/Event.php (hunk 2 above)
Comment (bordoni) [inline src/Tribe/Aggregator/Event.php]: `esc_sql` does that.

Comment (bordoni) [inline src/Tribe/Aggregator/Event.php]: Actually no, it's `$wpdb->prepare` that does that.

**Comment by barryhughes** [comment]
:noice:

**Pull request merged**